mds/models.py

```diff
@@ -164,12 +159,16 @@ def __str__(self):
         )
 
     @property
-    def latest_event(self):
-        if hasattr(self, "_latest_event"):
+    def latest_events(self):
+        if hasattr(self, "_latest_events"):
             # don't do a query in this case, the telemetry was prefetched.
-            return self._latest_event[0] if self._latest_event else None
-        device = Device.objects.filter(pk=self.pk).with_latest_event().get()
-        return device.latest_event
+            return self._latest_events[:5]
+        latestEvents = (
+            EventRecord.objects.filter(device_id=self.id)
+            .exclude(event_type="telemetry")
+            .order_by("-timestamp")[:5]
```

Comment:
Why 5 when you only want the last one? I know you want 5 on the frontend but you're introducing application requirements in this generic API. Make it a function parameter for when you call it on line https://github.com/Polyconseil/django-mds/pull/73/files#diff-0885fd11fed58baa6da2a3a2129ff488R125 ?

Reply:
yes going back to 1 event